protocol/x/clob/keeper/liquidations.go

```diff
@@ -29,9 +29,21 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
 ) error {
 	lib.AssertCheckTxMode(ctx)
 
+	pseudoRand := k.GetPseudoRand(ctx)
+
 	// Get the liquidation order for each subaccount.
+	// Process at-most `MaxLiquidationOrdersPerBlock` subaccounts, starting from a pseudorandom location
+	// in the slice.
 	liquidationOrdersForSorting := make([]types.LiquidationOrder, 0)
-	for _, subaccountId := range subaccountIds {
+	numSubaccounts := len(subaccountIds)
+	numLiqOrders := lib.Min(numSubaccounts, int(k.MaxLiquidationOrdersPerBlock))
+	indexOffset := 0
+	if numSubaccounts > 0 {
```

Comment:
For my understanding why not just early return if `numSubaccounts == 0`?

IMO that might make sense to early return because currently the stat `numFilledLiquidations` will be 0 if there are no liquidatable subaccounts (which makes sense but IMO that stat should be used for verifying percent of filled liquidation orders).

Reply:
Yeah just think about this more in general when you do metrics. I also noticed we always time `k.SortLiquidationOrders` even when there are no orders to sort